commerce/domain/tax/cart/calculator/classes/TaxCartCalculatorSample.cls

```diff
@@ -146,10 +164,40 @@ public class TaxCartCalculatorSample extends CartExtension.TaxCartCalculator {
                 cartItem.getCartItemPriceAdjustments(),
                 cipaTax.getId()
               );
-              adj.getCartTaxes().add(promoTax);
+              if (adj != null) {
+                adj.getCartTaxes().add(promoTax);
+              }
             }
         }
       }
+
+      // If there are shipping items, add tax for them as well
```

Comment:
How are you planning on testing this? Would you be modifying the Apex test class?

Reply:
Looks like you may need to modify the test cases to test this scenario

Reply:
We do not have any tests for this currently. I tested the use cases manually on steam org. We plan to raise a work item to cover the tests.

Reply:
Please add fTests as appropriate and Unit tests for this apex class.